### PR TITLE
feat(HACBS-1312): add unit testing for integration API

### PR DESCRIPTION
**Pull request by hongweiliu17**
Signed-off-by: Hongwei Liu hongliu@redhat.com

### CODE REVIEW

File: api/v1alpha1/integrationtestscenario_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+const (
+	// Timeout for Eventually blocks
+	timeout = time.Second * 10
+)
+
+func TestControllerSnapshot(t *testing.T) {
```
Comment (dirgim) [inline api/v1alpha1/integrationtestscenario_suite_test.go:46]: We probably want to call this function something else like `TestIntegrationAPIs`. 
I realize that I also copied this same name in [gitops_suite_test.go](https://github.com/redhat-appstudio/integration-service/blame/main/gitops/gitops_suite_test.go#L52).

File: api/v1alpha1/integrationtestscenario_types_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		}
+	})
+
+	AfterEach(func() {
+		err := k8sClient.Delete(ctx, integrationTestScenario)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	Context("When a integrationTestScenario is created", func() {
+		It("should be able to create integrationTestScenario", func() {
```
Comment (amisstea) [inline api/v1alpha1/integrationtestscenario_types_test.go:81]: I think we can drop this `Context` wrapper. The `It` description is informative enough and a little clearer, IMO.

Comment (hongweiliu17) [reply api/v1alpha1/integrationtestscenario_types_test.go]: Yes, thanks. Updated.

File: api/v1alpha1/integrationtestscenario_types_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			Expect(k8sClient.Create(ctx, integrationTestScenario)).Should(Succeed())
+			Expect(integrationTestScenario.Spec.Application).To(Equal("application-sample"))
```
[the remark this answers is not in the document]
Comment (hongweiliu17) [reply api/v1alpha1/integrationtestscenario_types_test.go]: Thanks. Updated

File: api/v1alpha1/integrationtestscenario_suite_test.go (hunk 1 above)
Comment (hongweiliu17) [reply api/v1alpha1/integrationtestscenario_suite_test.go]: Yes.. I updated them together

File: tekton/tekton_suite_test.go [hunk 4]
```diff
@@ -46,7 +46,7 @@ var (
 	cancel    context.CancelFunc
 )
 
-func TestControllerSnapshot(t *testing.T) {
+func TestTeckton(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (hongweiliu17) [reply tekton/tekton_suite_test.go]: Thanks!

**Review by dirgim — APPROVED** [review]
LGTM

**Review by srivickynesh — APPROVED** [review]
LGTM

**Review by amisstea — APPROVED** [review]
LGTM :+1:

**Pull request merged**